### PR TITLE
Mutable rings

### PR DESCRIPTION
**Pull request by tthsqe12**

### CODE REVIEW

**Comment by thofma** [comment]
Amazing, thanks!

File: src/generic/LaurentPoly.jl [hunk 1]
```diff
@@ -417,7 +419,8 @@ end
 ################################################################################
 
 function map_coefficients(f, p::LaurentPolyWrap)
```
Comment (thofma) [inline src/generic/LaurentPoly.jl:421]: I think this also should get a `cached` keyword, similar to the method for polynomials.

**Comment by tthsqe12** [comment]
Well, this last one is a mess. I need the `_change_poly_ring`, `_make_parent` and `_map` methods defined in `src/Poly.jl`.

File: test/generic/LaurentPoly-test.jl [hunk 2]
```diff
@@ -204,20 +218,27 @@ using AbstractAlgebra.Generic: Integers, LaurentPolyWrapRing, LaurentPolyWrap,
...
+   end
+
    @testset "comparisons" begin
       L, y = LaurentPolynomialRing(ZZ, "y")
       x = y.poly
 
       @test y == y
 
-      f = LaurentPolyWrap(x^3 + 2x^2 - 1)
+      f = LaurentPolyWrap(L, x^3 + 2x^2 - 1)
```
Comment (thofma) [inline test/generic/LaurentPoly-test.jl:234]: Just for clarification. This is the same as `L(x^3 + 2x^2 - 1)`?

Comment (tthsqe12) [reply test/generic/LaurentPoly-test.jl:234]: Yeah. We should probably change that. I dont like so many internal fxns in the test code.

Comment (thofma) [reply test/generic/LaurentPoly-test.jl]: Changed it.

**Pull request merged**